api/indexer/indexer_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package indexer
+
+import (
+	"testing"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/chain/chaintest"
+	"github.com/ava-labs/hypersdk/fees"
+)
+
+func TestBlockIndex(t *testing.T) {
```

Comment:
This test feels like it's testing a lot of different cases which is making it complex to read/debug/add to. I feel like testing more targeted code edge-cases instead of a single god-test would make sense here.

Reply:
Moved a helper to generate a sequence of empty executed blocks to `chaintest` and separated out the restart portion of the test.